### PR TITLE
ref(ts) Convert Inbound filters to typescript

### PR DESCRIPTION
**Pull request by markstory**
I have a chart to replace on this view so it makes sense to move to typescript first. I had to extend the `FormField` type to reflect load bearing real usage. The `getValue` prop ends up being consumed by FormModel in the `saveFieldRequest` method.

### CODE REVIEW

**Review by billyvg — APPROVED** [review]

File: src/sentry/static/sentry/app/views/settings/project/projectFilters/projectFiltersSettings.tsx [hunk 1]
```diff
@@ -163,37 +183,51 @@ class LegacyBrowserFilterRow extends React.Component {
...
 
-  handleLegacyChange = (onChange, onBlur, _filter, subfilters, e) => {
-    onChange(subfilters, e);
-    onBlur(subfilters, e);
+  handleLegacyChange = (
+    onChange: FormFieldProps['onChange'],
+    onBlur: FormFieldProps['onBlur'],
+    _filter,
+    subfilters: RowState['subfilters'],
+    e
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/settings/project/projectFilters/projectFiltersSettings.tsx:227]: shouldn't we set some type here? maybe we can use the Toggle event type?

File: src/sentry/static/sentry/app/views/settings/project/projectFilters/projectFiltersSettings.tsx [hunk 2]
```diff
@@ -68,7 +68,27 @@ const LEGACY_BROWSER_SUBFILTERS = {
 
 const LEGACY_BROWSER_KEYS = Object.keys(LEGACY_BROWSER_SUBFILTERS);
 
-class LegacyBrowserFilterRow extends React.Component {
+type FormFieldProps = React.ComponentProps<typeof FormField>;
+
+type RowProps = {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/settings/project/projectFilters/projectFiltersSettings.tsx:73]: shouldn't we name it `Props`?

File: src/sentry/static/sentry/app/views/settings/project/projectFilters/projectFiltersSettings.tsx [hunk 3]
```diff
@@ -68,7 +68,27 @@ const LEGACY_BROWSER_SUBFILTERS = {
...
+  };
+  onToggle: (
+    data: RowProps['data'],
+    filters: RowState['subfilters'],
+    event: React.MouseEvent
+  ) => void;
+  disabled?: boolean;
+};
+
+type RowState = {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/views/settings/project/projectFilters/projectFiltersSettings.tsx:85]: and `State` here?

File: src/sentry/static/sentry/app/views/settings/project/projectFilters/projectFiltersSettings.tsx (hunk 2 above)
Comment (markstory) [reply src/sentry/static/sentry/app/views/settings/project/projectFilters/projectFiltersSettings.tsx:73]: There are two components in this file, a row and container. I used `Props` and `State` for the container which is lower in the file (line 186)

File: src/sentry/static/sentry/app/views/settings/project/projectFilters/projectFiltersSettings.tsx (hunk 3 above)
Comment (markstory) [reply src/sentry/static/sentry/app/views/settings/project/projectFilters/projectFiltersSettings.tsx:85]: There are two components in this file, a row and container. I used `Props` and `State` for the container which is lower in the file (line 186)

**Pull request merged**